File: examples/docs-code-snippets/src/main/java/examples/java/interactions/ShipOrderImpl.java
package examples.java.interactions;

import examples.java.ingredients.Address;

public class ShipOrderImpl implements ShipOrder {
    @Override
    public OrderShipped apply(String orderId, Address address) {
        System.out.printf("Shipping order %s to %s", orderId, address);
        return new OrderShipped();
    }
}

File: examples/docs-code-snippets/src/main/scala/examples/scala/interactions/ShipOrderImpl.scala
package examples.scala.interactions

import examples.scala.ingredients.Address

import scala.concurrent.Future

trait ShipOrderTrait {
  def apply(orderId: String, address: Address): Future[ShipOrder.OrderShipped]
}

class ShipOrderImpl extends ShipOrderTrait {
  override def apply(orderId: String, address: Address): Future[ShipOrder.OrderShipped] = {
    println(s"Shipping order $orderId to $address")
    Future.successful(ShipOrder.OrderShipped())
  }
}
